File: Exemplos M10/App_IntentDigitalHUB_XamarinForms/Elgin_IntentDigitalHUB_XamarinForms-font/E1_Digital_Hub_Xamarin_Forms_M8/Balance/BalanceProtocol.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace Xamarin_Forms_Intent_Digital_Hub.Balance
{
    class BalanceProtocol
    {
        private BalanceProtocol(string friendlyName) { this.friendlyName = friendlyName; }

        private String friendlyName;

        public string toString() { return this.friendlyName; }

        public static BalanceProtocol PROTOCOL_0 = new BalanceProtocol("PROTOCOL 0");
        public static BalanceProtocol PROTOCOL_1 = new BalanceProtocol("PROTOCOL 1");
        public static BalanceProtocol PROTOCOL_2 = new BalanceProtocol("PROTOCOL 2");
        public static BalanceProtocol PROTOCOL_3 = new BalanceProtocol("PROTOCOL 3");
        public static BalanceProtocol PROTOCOL_4 = new BalanceProtocol("PROTOCOL 4");
        public static BalanceProtocol PROTOCOL_5 = new BalanceProtocol("PROTOCOL 5");
        public static BalanceProtocol PROTOCOL_6 = new BalanceProtocol("PROTOCOL 6");
        public static BalanceProtocol PROTOCOL_7 = new BalanceProtocol("PROTOCOL 7");
    }
}

File: Exemplos M10/App_IntentDigitalHUB_Kotlin/Elgin_IntentDigitalHUB_Kotlin-font/app/src/main/java/com/elgin/intent_digital_hub/Activities/Balanca/BalanceProtocol.kt
package com.elgin.intent_digital_hub.Activities.Balanca

/**
 * Protocolos de comunicação com a balança, o código de cada protocolo corresponde ao índice de declaração no enum
 * Ex: Protocolo 4 possui 4 de código correspondente
 */
enum class BalanceProtocol(val friendlyName: String) {
    PROTOCOL_0("PROTOCOL 0"),
    PROTOCOL_1("PROTOCOL 1"),
    PROTOCOL_2("PROTOCOL 2"),
    PROTOCOL_3("PROTOCOL 3"),
    PROTOCOL_4("PROTOCOL 4"),
    PROTOCOL_5("PROTOCOL 5"),
    PROTOCOL_6("PROTOCOL 6"),
    PROTOCOL_7("PROTOCOL 7");

    //Nome legível, utilizado para a montagem do Spinner/Dropdown
    override fun toString(): String {
        return this.friendlyName
    }
}